jwst/assign_wcs/util.py

```diff
@@ -88,6 +89,70 @@ def _reproject(x, y):
     return _reproject
 
 
+def compute_scale_transform(refwcs, fiducial):
+    """Compute scaling transform
+    """
+    x0, y0 = refwcs.backward_transform(*fiducial)
+
+    x1 = x0 + 1
+    y1 = y0 + 1
+    ra0, dec0 = refwcs(x0, y0)
+    ra_xdir, dec_xdir = refwcs(x1, y0)
+    ra_ydir, dec_ydir = refwcs(x0, y1)
```

Comment:
Could you put the inputs in an array and do the transformations once?

Reply:
Actually I'm not sure. For computing the "initial" points, I think that makes sense, but is there a way to do that with the `WCS` object without splitting an "initial point" array and "delta points" array to provide the appropriate inputs (x1, y0 and x0, y1)? I'm not very familiar with the `WCS` API yet, so if you had something in mind, let me know!

Reply:
Something like this can work:
```
crpix = refwcs.backward_transform(*fiducial)
crpix_with_offsets = np.vstack((crpix, crpix + (0, 1), crpix + (1, 0))).T
crval_with_offsets = refwcs(*crpix_with_offsets)

coords = SkyCoord(ra=crval_with_offsets[0], dec=crval_with_offsets[1], unit="deg")
xscale = np.abs(coords[0].separation(coords[1]).value)
yscale = np.abs(coords[0].separation(coords[2]).value)
```

Reply:
The wcs can operate on a numpy array.  So `wcs(x, y)` where x and y are each arrays of x values and y values.  That `vstack` and transpose manipulate the array to have that form so it can be indexed by 0 and 1.  This assumes that `fiducial` is a tuple of (RA, Dec), which is I think correct for this function?

Reply:
Got it, thanks @jdavies-st 

Also, yes, I believe that's correct

Reply:
Oh, it looks like I flipped `xscale` and `yscale` above.  WCS takes x first then y, so an offset (0, 1) is an offset in y, not x.